src/Form/Input.js

```diff
@@ -238,28 +300,50 @@ class Input extends Component {
       placeholder,
       transformOnBlur,
       size,
+      leftIcon,
+      leftIconSize,
+      leftIconColor,
+      onLeftIconClick,
+      rightIcon,
+      rightIconSize,
+      rightIconColor,
+      onRightIconClick,
       ...rest
     } = this.props;
 
-    const isFloating = (floating && value && value.length > 0) || (floating && this.state.value);
+    const { focused, height, value: stateValue } = this.state;
+
+    const isFloating = (floating && value && value.length > 0) || (floating && stateValue);
 
     const inputProps = {
       ...rest,
       id,
       ref: this.ref,
       size,
-      value: this.state.value,
+      value: stateValue,
       onChange: this.onChange,
       onFocus: this.onFocus,
       onBlur: this.onBlur,
-      style: multiline ? { ...style, height: this.state.height } : style,
+      style: multiline ? { ...style, height } : style,
       placeholder,
       isFloatable: floating,
       isFloating,
+      error,
+      leftIcon,
+      leftIconSize,
```

Comment:
I don't think these need to be passed to the actual input element

Reply:
We need a way to pass the `iconSize` to add the appropriate padding to either side of the input if either `icon` is present. 

I suppose we could turn the `leftIcon` into `hasLeftIcon: !!leftIcon` to reduce the explicit `hasLeftIcon={!!leftIcon} hasRightIcon={!!rightIcon}` passed to the input.

That's still passing them through though. :/

Reply:
Makes sense - a little verbose but that seems fine